packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -721,73 +1050,110 @@ All @auth directives used on field definitions are performed when the field is r
+                    field ? field.name.value : undefined
+                )
+
+                const collectAuthCondition = this.resources.collectAuthCondition()
+                const ifNotStaticallyAuthedCreateAuthCondition = iff(
+                    raw(`! $${ResourceConstants.SNIPPETS.IsStaticGroupAuthorizedVariable}`),
+                    compoundExpression([
+                        dynamicGroupAuthorizationExpression,
+                        newline(),
+                        ownerAuthorizationExpression,
+                        newline(),
+                        collectAuthCondition
+                    ])
+                )
 
-            const collectAuthCondition = this.resources.collectAuthCondition()
-            const ifNotStaticallyAuthedCreateAuthCondition = iff(
-                raw(`! $${ResourceConstants.SNIPPETS.IsStaticGroupAuthorizedVariable}`),
-                compoundExpression([
-                    dynamicGroupAuthorizationExpression,
-                    newline(),
-                    ownerAuthorizationExpression,
-                    newline(),
-                    collectAuthCondition
-                ])
-            )
+                const throwIfNotStaticGroupAuthorizedOrAuthConditionIsEmpty = this.resources.throwIfNotStaticGroupAuthorizedOrAuthConditionIsEmpty()
 
-            const throwIfNotStaticGroupAuthorizedOrAuthConditionIsEmpty = this.resources.throwIfNotStaticGroupAuthorizedOrAuthConditionIsEmpty()
+                const authModesToCheck = new Set<AuthProvider>();
+                const expressions: Array<Expression> = new Array();
 
-            const authorizationLogic = compoundExpression([
-                staticGroupAuthorizationExpression,
-                newline(),
-                ifNotStaticallyAuthedCreateAuthCondition,
-                newline(),
-                throwIfNotStaticGroupAuthorizedOrAuthConditionIsEmpty
-            ]);
+                if (ownerAuthorizationRules.find((r) => r.provider === 'userPools') ||
+                    staticGroupAuthorizationRules.length > 0 ||
+                    dynamicGroupAuthorizationRules.length > 0) {
+                    authModesToCheck.add('userPools');
+                }
+                if (ownerAuthorizationRules.find((r) => r.provider === 'oidc')) {
+                    authModesToCheck.add('oidc');
+                }
 
-            const templateParts = [
-                print(
-                    field && ifCondition ?
-                        iff(ifCondition, authorizationLogic) :
-                        authorizationLogic
-                ),
-                resolver.Properties.RequestMappingTemplate
-            ]
-            resolver.Properties.RequestMappingTemplate = templateParts.join('\n\n')
-            ctx.setResource(resolverResourceId, resolver)
+                if (authModesToCheck.size > 0) {
+                    expressions.push (this.resources.getAuthModeDeterminationExpression(authModesToCheck));
```

Comment:
Do you mean to be calling `getAuthModeDeterminationExpression` twice here and again on line 1143?

Reply:
No, on line 1143 I'm calling ```getAuthModeCheckWrappedExpression```. The new resolver logic has 2 parts:
- determine the authMode from the request
- wrap the existing owner/group etc checks into an ```if``` block if authMode is not matching the one we need compare against.

in 1131 I'm adding the authMode check to the start of the resolver, later the ```if``` block.